src/sdi-snap.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include "sdi-refresh-dialog.h"
+#include <snapd-glib/snapd-glib.h>
+
+G_BEGIN_DECLS
+
+#define SDI_TYPE_SNAP sdi_snap_get_type()
```

Comment:
Nit: I have not bothered to make these defines - you can just use the _get_type() function directly (as you have).

Reply:
Yes, but for consistency with GObject I think that it's a good idea to use them.

Reply:
(mainly for the cases where properties are used).